state/immutable-trie/state.go

```diff
@@ -27,6 +27,19 @@ func NewState(storage Storage) *State {
 }
 
 func (s *State) NewSnapshot() state.Snapshot {
+	t := s.newTrie()
+	return &Snapshot{state: s, trie: t}
```

Comment:
Minor thing:
We don't need trie reference (unless you've left it for the debugging purposes)? 🙂 

```suggestion
	return &Snapshot{state: s, trie: s.newTrie()}
```